core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java

```diff
@@ -398,6 +403,42 @@ public void getBlockInfo() throws Exception {
     assertEquals(expectedBlockInfo, mBlockMaster.getBlockInfo(blockId));
   }
 
+  @Test
+  public void getNewContainerId() throws Exception {
+    int total = 10000;
+    long containerIdReservationSize = Configuration.getInt(
+        PropertyKey.MASTER_CONTAINER_ID_RESERVATION_SIZE);
+
+    int parallelNum = 5;
+    ExecutorService containerIdFetcher = Executors.newFixedThreadPool(parallelNum);
+    AtomicInteger times = new AtomicInteger(0);
+    assertEquals(0L, mBlockMaster.getNewContainerId());
+
+    CyclicBarrier barrier = new CyclicBarrier(parallelNum);
+
+    for (int count = 0; count < parallelNum; count += 1) {
+      containerIdFetcher.submit(() -> {
+        try {
+          barrier.await();
+          while (times.incrementAndGet() < total) {
+            mBlockMaster.getNewContainerId();
+          }
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      });
+    }
+
+    containerIdFetcher.shutdown();
+    containerIdFetcher.awaitTermination(6, TimeUnit.SECONDS);
+    Thread.sleep(1000);
```

Comment:
Could you remove this sleep? It does not seem to affect the success or failure of this test, and multiplies the test's runtime by 100. 
```suggestion
```

Reply:
"Thread.sleep(1000)" is replaced with "mBlockMaster.close()",  What it does is let potential background task in mContainerIdDetector complete